core/src/replay_stage.rs

```diff
@@ -4921,6 +4928,179 @@ pub mod tests {
         assert!(progress.is_propagated(root_bank.slot()));
     }
 
+    #[test]
+    fn test_dead_slot_then_multiple_epoch_slots_frozen_signals() {
+        /*
+            Build fork structure:
+
+                  slot 0
+                    |
+                  slot 1
+                 /      \
+        slot 2 (dead)    |
+                      slot 3 (dead)
+        */
+
+        let tree = tr(0) / tr(1);
+        let ReplayBlockstoreComponents {
+            rpc_subscriptions,
+            blockstore,
+            vote_simulator,
+            ..
+        } = replay_blockstore_components(Some(tree), 1, None::<GenerateVotes>);
+
+        let VoteSimulator {
+            mut progress,
+            bank_forks,
+            ..
+        } = vote_simulator;
+
+        // Create situation where slots with *common ancestors* are marked dead,
+        // but then we detect the rest of the cluster has frozen the slot via EpochSlots.
+        // There should not be multiple repair requests generated for the common ancestors.
```

Comment:
Could there still be multiple requests due to dump and repair trying to fix these at the same time?

Reply:
`duplicate_slot_repair_request_sender` is a hashmap so should de-duplicate any outstanding requests. 

Once the request has been made, however, and the slot removed from `duplicate_slot_repair_request_sender`, we may add it again, but only if the bank has been frozen again (another wrong hash for instance).